### PR TITLE
[runtime] Add MONO_EMPTY_SOURCE_FILE to a couple of files.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by jaykrell** [comment]
I’d remove the else & make it unconditional but ok either way.

File: mono/utils/mono-dl-darwin.c [hunk 1]
```diff
@@ -63,4 +63,10 @@ mono_dl_get_system_dir (void)
 #endif
 }
 
+#else
+
+#include <mono/utils/mono-compiler.h>
+
+MONO_EMPTY_SOURCE_FILE (mono_dl_darwin);
```
Comment (jaykrell) [inline mono/utils/mono-dl-darwin.c]: I’d omit the semicolons also but ok.

File: mono/mini/aot-runtime.c [hunk 2]
```diff
@@ -187,8 +187,9 @@ static GHashTable *static_aot_modules;
  * and it needs to be fully loaded by the time the other code needs it, it
  * must be eagerly loaded before other modules.
  */
-static char *container_assm_name = NULL;
-static MonoAotModule *container_amodule = NULL;
+static char *container_assm_name;
```
Comment (jaykrell) [inline mono/mini/aot-runtime.c]: Seems unrelated.

File: mono/mini/aot-runtime.c [hunk 3]
```diff
@@ -301,9 +302,14 @@ load_image (MonoAotModule *amodule, int index, MonoError *error)
 	 * current AOT module matches the wanted name and guid and just return
 	 * the AOT module's assembly.
 	 */
-	if (!strcmp (amodule->assembly->image->guid, amodule->image_guids [index]))
+	if (!strcmp (amodule->assembly->image->guid, amodule->image_guids [index])) {
```
Comment (jaykrell) [inline mono/mini/aot-runtime.c]: Seems unrelated, might or might not be, ok.

File: mono/metadata/w32process-unix-osx.c [hunk 4]
```diff
@@ -171,4 +171,8 @@ mono_w32process_get_modules (pid_t pid)
 	return g_slist_reverse (ret);
 }
 
+#else
+
+MONO_EMPTY_SOURCE_FILE (w32process_unix_bsd);
```
Comment (jaykrell) [inline mono/metadata/w32process-unix-osx.c]: Bsd vs. osx.

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/aot-runtime.c (hunk 2 above)
Comment (jaykrell) [reply mono/mini/aot-runtime.c]: I do prefer this style though. Terser and perfectly clear.

**Comment by vargaz** [comment]
@monojenkins build failed
